### ISSUE TITLE
Cannot get minimal ECS sample to work

### ISSUE THREAD

**Issue by Ithamar** [3mcd/javelin#150]
I really like the design of this API, it looks really great. However, when I try to build a minimal sample code, I can't seem to get it to work. It is probably my fault, but I'm at a loss as to why

```js
const {
  createComponentType,
  createWorld,
  onAttach,
  onDetach,
  query,
} = require('@javelin/ecs')

const Test = createComponentType({ type: 1 })

const testQ = query(Test)

function system() {
  console.log('In system')

  onAttach(Test).forEach(e => {
    console.log(`Entity ${e} created`)
  })
  onDetach(Test).forEach(e => {
    console.log(`Entity ${e} destroyed`)
  })

  for (const [entities] of testQ) {
    console.log(`testQ: found ${entities.length} entities`)
  }
}

const w = createWorld({
  systems: [ system ]
});

w.spawn(Test)
w.spawn(Test)

setInterval(w.tick, 1000)
```

The only output I see is "In System" every seconds, but onAttach/onDetach/query does not seem to produce any results.

Any suggestions?
(This was tested with v0.21.0)

**Comment by 3mcd** [comment]
Hey @Ithamar, I'm glad you like the API. Could you give this a shot?

```ts
w.spawn(w.component(Test))
w.spawn(w.component(Test))
```

**Comment by 3mcd** [comment]
Also – I think your example does bring another issue to light with `onAttach`/`onDetach`. They don't seem to be triggering for entities that are created prior to the first tick. I'll get to work on a fix :)

**Comment by Ithamar** [comment]
That did it, thanks a lot for the super-quick response! I already tried calling tick before spawn, but without the world.component() thing it did not help much ;)

But feel free to do anything you like with this code, it was just meant to make it easier to see where I went wrong.

**Comment by 3mcd** [comment]
@Ithamar FYI the issue you discovered (where triggers don't yield components attached/detached prior to the first call) was fixed in 0.21.2. https://runkit.com/emcd/onattach-ondetach

**Comment by Ithamar** [comment]
great, means I can upgrade and remove that world.tick() I inserted ;)

I'm sorry to do this here, but is there any chance you could take comment on 3mcd/javelin-fps#3 & 3mcd/javelin-fps#4  too? I'm guessing it would only take a minute to answer ;)
